mypy/build.py

```diff
@@ -2315,6 +2326,36 @@ def find_module_and_diagnose(manager: BuildManager,
             raise CompileError(["mypy: can't find module '%s'" % id])
 
 
+def in_partial_package(id: str, manager: BuildManager) -> bool:
+    """Check if a missing module can potentially be a part of a package.
+
+    This checks if there is any existing parent __init__.pyi stub that
+    defines a module-level __getattr__ (a.k.a. partial stub package).
+    """
+    while '.' in id:
+        parent, _ = id.rsplit('.', 1)
+        parent_mod = None
```

Comment:
Maybe move this into the `except` block (L2345)? I was looking whether parent_mod is always defined (because it's not set in the `except` clause) and this felt too far back up. A different way to structure this would be `parent_mod = manager.modules.get(parent)` followed by `if parent_mod is None`.